integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/OpenLineageRunEventBuilder.java

```diff
@@ -206,12 +210,13 @@ RunEvent buildRun(
     nodes.addAll(Rdds.flattenRDDs(rdd));
 
     return populateRun(
-        applicationParentRunFacet, runEventBuilder, jobBuilder, jobFacetsBuilder, nodes);
+        applicationParentRunFacet, runEventBuilder, runId, jobBuilder, jobFacetsBuilder, nodes);
   }
 
   RunEvent buildRun(
       ParentRunFacet applicationParentRunFacet,
       RunEventBuilder runEventBuilder,
+      UUID runId,
```

Comment:
Isn't runId present within OpenLineageContext? 

How about creating a separate `OpenLineageApplicationEventBuilder` covered this time with unit-tests? Would this make sense?

Reply:
OpenLineageContext has getRunUuuid(), but for Spark Application events we need `eventEmitter.getApplicationId()`, which is used for Spark job events as parentRunId. Should I add `eventEmitter` field to this class?

Regarding `OpenLineageApplicationEventBuilder`, this can lead to a bit of duplication, e.g. `buildJobFacets` and `buildRunFacets`. Is that okay?

Reply:
`OpenLineageApplicationEventBuilder` could extend common abstract class as `OpenLineageRunEventEventBuilder` to implement common methods. 

Similar to what you propose would be adding `applicationId` field to `OpenLineageContext` which should be also fine I think. 

I just would like to avoid having two different `runId` in `OpenLineageRunEventEventBuilder`, having different meanings and being passed in a different way.

Reply:
Added `applicationUuid` to `OpenLineageContext`